kdbg/debugger.cpp

```diff
@@ -2302,9 +2281,8 @@ void KDebugger::handleSetDisassFlavor(const char* output)
 	return;
     }
 
-    bool isX86 = isTargetX86();
-    if (isX86) {
-	emit asmFlavorChangedForTarget(m_flavor, m_target);
+    if (isTargetX86()) {
+	emit asmFlavorChangedForTarget(enumToFlavor(m_flavor), m_target);
```

Comment:
Hmm, now I see that this `isTargetX86()` does not work anymore as you intended because it looks only at the *This Program* setting, not at the effective setting that was submitted. Perhaps it's indeed necessary to store the submitted value in another member variable, say, `m_effectiveFlavor`.